esp/src/src-react/components/forms/RenameFile.tsx

```diff
@@ -38,42 +42,71 @@ export const RenameFile: React.FunctionComponent<RenameFileProps> = ({
-                            const request = { ...data, srcname: logicalFile, dstname: data.targetRenameFile[idx].name };
-                            const requests = [];
-                            requests.push(FileSpray.Rename({ request: request }));
-                            Promise.all(requests).then(_ => {
-                                closeForm();
-                                if (refreshGrid) refreshGrid(true);
-                            });
-                        });
+            async (data, evt) => {
+                const renameRequests = [];
+                const getDfuWuRequests = [];
+
+                if (logicalFiles?.length === 1) {
+                    const request = { ...data, srcname: logicalFiles[0], DFUServerQueue: "" };
+                    renameRequests.push(service.Rename(request));
+                } else {
+                    logicalFiles.forEach((logicalFile, idx) => {
+                        const request = { ...data, srcname: logicalFile, dstname: data.targetRenameFile[idx].name, DFUServerQueue: "" };
+                        renameRequests.push(service.Rename(request));
+                    });
+                }
+
+                const renameResponses = await Promise.all(renameRequests);
+                renameResponses.forEach(response => {
+                    const wuid = response?.wuid ?? null;
+                    if (wuid) {
+                        getDfuWuRequests.push(service.GetDFUWorkunit({ wuid }));
+                    }
+                });
+
+                const getDfuWuResponses = await Promise.all(getDfuWuRequests);
+                if (getDfuWuResponses.length === 1) {
+                    const { ID = null, State = 0, SummaryMessage = "" } = { ...getDfuWuResponses[0]?.result };
+                    if (State === FileSprayStates.failed) {
+                        logger.error(SummaryMessage ?? "");
+                        if (ID) {
+                            pushUrl(`/dfuworkunits/${ID}`);
+                        }
+                    }
+                    if (ID) {
+                        if (window.location.hash.match(/#\/files\//) === null) {
+                            pushUrl(`/dfuworkunits/${ID}`);
+                        } else {
+                            replaceUrl(`/dfuworkunits/${ID}`);
+                        }
                     }
+                } else {
+                    getDfuWuResponses.forEach(response => {
+                        const State = response?.result?.State ?? FileSprayStates.unknown;
+                        const ID = response?.result?.ID;
+                        if (State === FileSprayStates.failed) {
+                            logger.error(response?.result?.SummaryMessage ?? "");
+                        } else if (ID) {
+                            window.setTimeout(function () { window.open(`#/dfuworkunits/${ID}`); }, 0);
+                        }
+                    });
```

Comment:
This if / else has different logic for error handling?  Could be combined to a single flow with the exception of the `window.open` and `pushUrl`?